src/main/java/touro/chess/BishopPiece.java

```diff
@@ -18,36 +18,36 @@ public BishopPiece(Location location, PieceColor color)
 //        int col = this.getLocation().getColumn();
 
         //Possible moves to the upper left
-        for(int row = this.getLocation().getRow(), col = this.getLocation().getColumn();
-            col >= 1; row--, col--)
+        for(int row = (this.getLocation().getRow()) - 1, col = (this.getLocation().getColumn()) - 1;
+            col >= 1 && row >= 1; row--, col--)
         {
             Location newLocation = new Location(row, col);
             moves.add(new Move(this.getLocation(), newLocation, false));
         }
 
-//        //Possible moves to the lower left
-//        for(int row = this.getLocation().getRow(), col = this.getLocation().getColumn();
-//            col >= 1; row++, col--)
-//        {
-//            Location newLocation = new Location(row, col);
-//            moves.add(new Move(this.getLocation(), newLocation, false));
-//        }
-//
-//        //Possible moves to the upper right
-//        for(int row = this.getLocation().getRow(), col = this.getLocation().getColumn();
-//            col <= 8; row--, col++)
-//        {
-//            Location newLocation = new Location(row, col);
-//            moves.add(new Move(this.getLocation(), newLocation, false));
-//        }
-//
-//        //Possible moves to the lower right
-//        for(int row = this.getLocation().getRow(), col = this.getLocation().getColumn();
-//            col <= 8; row++, col++)
-//        {
-//            Location newLocation = new Location(row, col);
-//            moves.add(new Move(this.getLocation(), newLocation, false));
-//        }
+        //Possible moves to the lower left
+        for(int row = (this.getLocation().getRow()) + 1, col = (this.getLocation().getColumn()) - 1;
+            col >= 1 && row <= 8; row++, col--)
```

Comment:
Didn't think to use both conditions in a for loop, this looks a lot neater than my while loops!